Adding weather effect sytem (#51)

GrumpyMurloc opened the pull request:
Added the Weather system in issue #51. Create a new formatter that return a message at the beginning of the day and the night determined by the attribute Weather contained in the class "Game". 

This is my first pull request, any comment are appreciated, either on code or git.


Signed-off-by: GrumpyMurloc <leblanc.nicolas@hotmail.com>

GrumpyMurloc commented on src/Game/Command/WeatherCommand.php:
```diff
@@ -27,7 +27,22 @@ public function fire()
 
-          $this->gameManager->sendMessageToChannel($this->game, ":rain_cloud: It is raining. It is a cold rain, and the freezing drops chill you to the bone." );
-
+        $weather= $this->game->weather;
```
Okay. Do I need to make another PR or is there a way to modify the current one ?

GrumpyMurloc commented:
I'll make the suggested modification as soon as possible. As for the reason why the message is sent only if the seer role is enable is because I  replaced the old message for the weather formatter and didn't realize this condition. This will be corrected in the next commit. 

Thanks for all the comments, I'm still learning and that help me a lot.

chrisgillis commented:
Awesome :) I think this project is great for learning PHP. While it's not particularly well written and it doesnt have unit tests or anything, it's still relatively simple PHP! 

Pull requests and code reviews should always be treated as learning opportunities, not just for the author of the code being reviewed but for the reviewer as well!

Thanks again for contributing :)

GrumpyMurloc commented:
Finished adding the modification ( finally got time to do it ). I changed a lot of on how the WeatherFormatter work. Right now weather is not an attribute of the game anymore. 

Added a weight system for the weather and deactivated the WeatherCommand. 
Added a Data folder containing the various messages.

chrisgillis commented on src/Game/Formatter/WeatherFormatter.php:
```diff
@@ -9,64 +9,114 @@
 
-        if($state == GameState::DAY){
+        if ($state == GameState::DAY){
+            if (file_exists("src/Game/Data/day_weather.txt")){
```
consider having these files load as part of the application boot process. this feels a little hidden away and maybe a little sneaky to me.

maybe encapsulate weather loading and retrieval in a Weatherman class and supply a new instance of it to the Game constructor, for example.
